presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -343,13 +350,32 @@ protected ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaData met
         if (mapping.isPresent()) {
             return mapping;
         }
-        Optional<ColumnMapping> connectorMapping = jdbcTypeToPrestoType(typeHandle);
-        if (connectorMapping.isPresent()) {
-            return connectorMapping;
+
+        mapping = jdbcTypeToPrestoType(typeHandle);
+        if (mapping.isPresent()) {
+            switch (legacyGenericColumnMapping) {
+                case ENABLE:
+                    return mapping;
+                case THROW:
+                    throw new IllegalStateException(format(
+                            "Column type %s is not explicitly mapped by the connector, and used to be mapped to %s by deprecated generic mappings. " +
+                                    "You can set '%s' to '%s' to temporarily restore the legacy mapping, or to '%s' to ignore the column.",
```

Comment:
`to ignore` is not exactly percise as the optional `CONVERT_TO_VARCHAR` is still in place. Would you rather return `Optional.empty()` instead break in `IGNORE` branch?

Reply:
> Would you rather return Optional.empty() instead break in IGNORE branch?

That would not be correct with respect to `unsupported_type_handling`.

Can you please help me improve the wording here?

Reply:
you can resplace `to ignore the column` with `to fallback to default handling of unsupported types` ?

Reply:
"to suppress this message" ?

Reply:
Even better :P

Reply:
Updated